### PR TITLE
Create boxes with click and drag

### PR DESCRIPTION
**Pull request by henriettemoe**
Create boxes by clicking and dragging (down and to the right)

### CODE REVIEW

File: src/components/Grid/Grid.tsx [hunk 1]
```diff
@@ -22,16 +24,75 @@ export const Grid: React.FC<GridProps> = ({
...
   const gridIndicatorLabel = "Click to create a new element";
 
+  const createBoxStart = React.useCallback((index: number) => {
+    setBoxStartPosition(index);
+  }, []);
+
+  const createBoxEnd = React.useCallback(
+    (indicatorIndex: number) => {
+      if (activeTool === ToolbarButtons.CreateBox) {
+        if (!boxStartPosition && boxStartPosition !== 0) {
```
Comment (boyum) [inline src/components/Grid/Grid.tsx:48]:
```suggestion
        if (boxStartPosition == null) {
```

File: src/components/Grid/Grid.tsx [hunk 2]
```diff
@@ -22,16 +24,75 @@ export const Grid: React.FC<GridProps> = ({
...
+
+        const xEnd = indicatorValue % numberOfColumns;
+        const xEndPercentagePosition = lastIndexOnColumn
+          ? 100
+          : (xEnd / numberOfColumns) * 100;
+
+        const widthPercentage = xEndPercentagePosition - xPercentagePosition;
+
+        // Create box
+        const id = (items.length + 1).toString();
```
Comment (boyum) [inline src/components/Grid/Grid.tsx:77]:
```suggestion
        const id = (items.length + 1).toString(); // TODO: Generate unique id
```

File: src/components/Grid/Grid.tsx [hunk 3]
```diff
@@ -22,16 +24,75 @@ export const Grid: React.FC<GridProps> = ({
...
+
+        const newItem = {
+          id: items ? id : "1",
```
Comment (boyum) [inline src/components/Grid/Grid.tsx:80]: Is the `items` null check necessary?

```suggestion
          id,
```

File: src/components/GridIndicator/GridIndicator.tsx [hunk 4]
```diff
@@ -3,18 +3,26 @@ import styles from "./GridIndicator.module.scss";
...
   label,
+  index,
 }) => {
   return (
     <button
       type="button"
       className={`grid-indicator ${styles.gridIndicator}`}
       onClick={onClick}
+      onMouseDown={() => onMouseDown(index)}
+      onMouseUp={() => onMouseUp(index)}
```
Comment (boyum) [inline src/components/GridIndicator/GridIndicator.tsx:25]: Should we add touch events as well?

```suggestion
      onMouseDown={() => onMouseDown(index)}
      onMouseUp={() => onMouseUp(index)}
      onTouchStart={() => onMouseDown(index)}
      onTouchEnd={() => onMouseUp(index)}
```

File: src/components/Grid/Grid.tsx (hunk 3 above)
Comment (henriettemoe) [reply src/components/Grid/Grid.tsx:80]: No, works with your suggestion 👍

**Review by boyum — APPROVED** [review]

**Pull request merged**